### PR TITLE
aws-cdk 1.7.0 (new formula)

### PR DESCRIPTION
**Pull request by nickholl**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

The AWS Cloud Development Kit is a framework for defining cloud infrastructure in code [https://docs.aws.amazon.com/cdk](url)

### CODE REVIEW

File: Formula/aws-cdk.rb
Comment (SMillerDev) [inline Formula/aws-cdk.rb]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Comment (nickholl) [reply Formula/aws-cdk.rb]: Apologies, I've added a (hopefully) valid test of functionality that doesn't require creds. My poor defence is that I was looking at other formula for inspiration and some of what I saw were version checks.

**Comment by nickholl** [comment]
@BrewTestBot test this please

**Comment by nickholl** [comment]
@SMillerDev how does this look now?

**Comment by zanhsieh** [comment]
@nickholl 
Do you think CDK would be faster to get approval by `brew bump-formula-pr`?
See example,
https://discourse.brew.sh/t/example-formula-update-pr-from-travis-ci/603/5

Also, a lot of closed PRs were auto-approved via `brew bump-formula-pr`.

**Comment by SMillerDev** [comment]
You can't bump a formula that didn't exist before and nothing gets auto approved.

**Comment by nickholl** [comment]
@zanhsieh I think that's a no. @SMillerDev is there anything further I can do or is it just a matter of patience at this point?

File: Formula/aws-cdk.rb
Comment (SMillerDev) [inline Formula/aws-cdk.rb]: This will break on every automated update. If you want to do a version check, please check for `#{version}`

File: Formula/aws-cdk.rb
Comment (SMillerDev) [inline Formula/aws-cdk.rb]: Please group the 2 tests together

File: Formula/aws-cdk.rb
Comment (nickholl) [reply Formula/aws-cdk.rb]: Done

File: Formula/aws-cdk.rb
Comment (nickholl) [reply Formula/aws-cdk.rb]: Could you elaborate?

**Comment by chenrui333** [comment]
@nickholl, thanks for your contributions, can you squash all the commits into one single commit? Thanks!

**Comment by nickholl** [comment]
Done

**Comment by zanhsieh** [comment]
@chenrui333 
Could you please check? Thank you.

**Review by chenrui333 — APPROVED** [review]
👍

**Comment by zanhsieh** [comment]
@nickholl
Since it's merged, could you bump to v1.9.0 please?

**Comment by chenrui333** [comment]
@zanhsieh Addressed in #44538.

**Pull request closed without merging**
